src/scaler/daemonscaler/provider.rs

```diff
@@ -193,25 +194,17 @@ impl<S: ReadStore + Send + Sync> ProviderDaemonScaler<S> {
     /// Construct a new ProviderDaemonScaler with specified configuration values
     pub fn new(store: S, config: ProviderSpreadConfig, component_name: &str) -> Self {
         let id = {
-            let default = format!(
-                "{PROVIDER_DAEMON_SCALER_TYPE}-{}-{component_name}-{}",
-                config.model_name, config.provider_id,
-            );
-
-            match &config.provider_config {
-                Some(provider_config) => {
-                    if let Some(provider_config_hash) =
-                        compute_provider_config_hash(provider_config)
-                    {
-                        format!(
-                            "{PROVIDER_DAEMON_SCALER_TYPE}-{}-{component_name}-{}-{}",
-                            config.model_name, config.provider_id, provider_config_hash
-                        )
-                    } else {
-                        default
-                    }
-                }
-                None => default,
+            if config.provider_config.is_empty() {
+                format!(
+                    "{PROVIDER_DAEMON_SCALER_TYPE}-{}-{component_name}-{}",
+                    config.model_name, config.provider_id,
+                )
+            } else {
+                let provider_config_hash = compute_provider_config_hash(&config.provider_config);
+                format!(
+                    "{PROVIDER_DAEMON_SCALER_TYPE}-{}-{component_name}-{}-{}",
+                    config.model_name, config.provider_id, provider_config_hash
+                )
```

Comment:
hmnnn feels like we could build this and then append the provider config hash if necessary?

Reply:
Yeah, I agree here. This is a nit, but I think you could do:

```
let mut id = format!(
                    "{PROVIDER_DAEMON_SCALER_TYPE}-{}-{component_name}-{}",
                    config.model_name, config.provider_id,
                );
if !config.provider_config.is_empty() {
    id.push('-');
    id.extend(compute_provider_config_hash(&config.provider_config))
}
```